src/features/instrument/Instrument.tsx

```diff
@@ -45,7 +38,52 @@ export const Instrument = () => {
     part.loopEnd = "4m";
   }
 
-  // @TODO: maybe move to separate components, maybe same file
+  // @TODO: check arent a beter way to write each thing now (always create new player?), also check updates in tone
+  function playSample(sampleName: string) {
+    new Player({
+      url: sampleName,
+      autostart: true,
+    }).chain(mediumVolume, Destination);
+  }
+
+  return (
+    <div className="Instrument">
+      <Keys></Keys>
+      <div className="PredefinedMusic">
+        <Melody melodyPart={melodyPart} playPart={playPart}></Melody>
+        <Bass bassPart={bassPart} playPart={playPart}></Bass>
+        <NoiseSound noise={noise}></NoiseSound>
+        <MarioSamples playSample={playSample}></MarioSamples>
+        <SpeedUpBpm></SpeedUpBpm>
+        <StopInstruments playSample={playSample} melodyPart={melodyPart} bassPart={bassPart} noise={noise}></StopInstruments>
+      </div>
+    </div>
+
+  );
+}
+
+export const Keys = () => {
+  function playNote(note: string) {
+    polySynthSquare.triggerAttackRelease(note, "8n");
+    sampler.triggerAttack(note);
+  }
+
+  return (
+    <div>
+      {GAMMA_NOTES.map(note => (
+        <Button onClick={() => playNote(note)} key={note} sx={keyStyles}>♪ {note}</Button>
+      ))}
+    </div>
+  );
+}
+
+//hmm slowly becomes a monster
```

Comment:
Haha, noted this during the creation of the first component - `Melody`. And by seeing the final result after moving all instruments it really looks like a monster :D I think to do not restructure in this way, I thought it would look better in my imagination before I started doing it. **WDYT?**

My thoughts:
+ It groups required pieces for that specific component, maybe can be easier to grasp the scope
- some parts are reusable, so ^ doesn't show full scope required for the specific instrument
- declarations of component (including props interfaces) drags attention from the logic inside
- return of Instrument component didn't become slimmer or more readable, `Button` line was just replaced with the line of a specific component 

I added all components in same file with an idea later maybe to move to separate files. Maybe that would be best if I plan to continue developing my component. But if I think that it's the final version, **does moving to separate files make sense?** Instruments feels too small, especially last ones. 

Lastly, for my presentation, I think components in separate files would provide extra confusion, because wouldn't let to see full picture and many jumps to different files. **Do you have another opinion on this?**